src/features/unified-share-modal/EmailForm.js

```diff
@@ -107,6 +136,23 @@ class EmailForm extends React.Component<Props, State> {
         }
     };
 
+    handleRemoveRestrictedExternalContacts = () => {
+        const { onContactRemove, selectedContacts, updateSelectedContacts } = this.props;
+
+        const [removedContacts, remainingContacts] = partition(selectedContacts, ({ value }) =>
+            this.isRestrictedExternalEmail(value),
+        );
+
+        updateSelectedContacts(remainingContacts);
```

Comment:
It wasn't possible to achieve this by calling `handleContactRemove` since that relies on the contact index, which might change each time a contact was removed.